inputremapper/injection/event_reader.py

```diff
@@ -60,7 +60,7 @@ def __init__(
         self,
         context: Context,
         source: evdev.InputDevice,
-        forward_to: evdev.UInput,
+        forward_to: evdev.UInput,  # TODO I think this can go as well
```

Comment:
And also the source. And provide the correct origin_hash instead.

Reply:
Thanks.

About the source: I'd generally prefer passing the original objects whenever possible, instead of derived information or members.

Reply:
```python
hash_a = "some hash"

context.get_source(hash_a)  # returns the source device
context.get_uinput(hash_a)  # returns the forward_to device
```
Since we need the origin hash to retrieve the correct forward_to device we can reduce the number of arguments to `notify()` by providing only the hash.

This also eliminates the possibility to create a bug where the `source` has a different `origin_hash` than the one provided. 

We could instead not provide the origin_hash but calculate it during runtime, but that is expensive.

Reply:
> Since we need the origin hash to retrieve the correct forward_to device we can reduce the number of arguments to notify() by providing only the hash.

the correct forward_to device is resolved by looking at the InputConfig objects though

Reply:
That is correct, so we don't even need the hash.

It is not relevant to this PR, but I was looking through all the `notify()` methods, and the source is only ever used to do some init stuff like finding out the `absinfo`.

Now that the context knows about all the sources and it gets passed to all `MappingHandlers` we should probably remove the `source` argument form `notify` and do the init stuff inside `__init__` that would clean up a few mapping handlers.